spatial/r3/rotation.go

```diff
@@ -0,0 +1,43 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/num/quat"
+)
+
+// TODO:
+//  - create rotations from Euler angles (NewRotationFromEuler?)
+//  - create rotations from rotation matrices (NewRotationFromMatrix?)
+//  - return the equivalent Euler angles from a Rotation
+//  - return the equivalent rotation matrix from a Rotation
+
+// Rotation describes a rotation in space.
+type Rotation struct {
+	q quat.Number
+}
```

Comment:
If this is `type Rotation quat.Number` then the system is more flexible and allows the user to do scaling and so on as well.

Reply:
SGTM.